components/IngredientDetails.js

```diff
@@ -0,0 +1,132 @@
+import { useRouter } from "next/router";
+import Image from "next/image";
+import styled from "styled-components";
+import ingredients from "@/assets/ingredients.json";
+import { useEffect, useState } from "react";
+
+const IngredientDetails = () => {
+  const flavorColors = {
+    Bitter: "#D2691E",
+    Creamy: "#FFFACD",
+    Fresh: "#98FB98",
+    Herbal: "#228B22",
+    Pungent: "#FF6347",
+    Salty: "#F0E68C",
+    Savory: "#D2B48C",
+    Spicy: "#cf2210",
+    Sour: "#d9df0c",
+    Sweet: "#FFB6C1",
+    Tangy: "#FFA07A",
+    Umami: "#188989",
+  };
+
+  const router = useRouter();
+  const { id } = router.query;
+  const [ingredient, setIngredient] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      const foundIngredient = ingredients.find(
+        (ingredient) => ingredient._id === id
+      );
+      setIngredient(foundIngredient);
+    }
+  }, [id]);
```

Comment:
You don't need a useEffect here and neither a state. This will work just as fine:

```suggestion
  const ingredient = ingredients.find((ingredient) => ingredient._id === id);
```